src/lib/dnssd/Resolver.h

```diff
@@ -62,13 +68,15 @@ struct ResolvedNodeData
     }
 
     PeerId mPeerId;
-    Inet::IPAddress mAddress               = Inet::IPAddress::Any;
-    Inet::InterfaceId mInterfaceId         = Inet::InterfaceId::Null();
+    int mNumIPs = 0;
+    Inet::InterfaceId mInterfaceId;
+    Inet::IPAddress mAddress[kMaxIPAddresses];
```

Comment:
@cecille My point is taht "devices with RAM challenges" == "a bunch of things in our CI".

As in, as part of this PR we should be making whatever tuning changes we need to minimize the hit on our RAM-constrained test apps.  Reducing the cache size for things like qpg and whatever other very-constrained platforms we have seems like a fine solution; it just needs to happen.

Reply:
I'm open to hearing ideas, but I think the reality is that if we're going to be caching a bunch of stuff, it's going to cost RAM and if we're only caching part of the information we get back, that cache is not useful. 

For CI, do we know approximately how many responses each node is receiving?

Reply:
The nodes that receive information in CI (on Linux and Darwin) are not the same as the things that are constrained devices (which we _compile_ in CI and measure the size of, and ensure can fit on the hardware, but don't actually _run_).

That said, on Linux/Mac I would expect us to receive somewhere around two responses: one for ipv4 and one for ipv6, both for the same node id.....

So concretely, I think we should leave the Linux/Darwin cache size as it is and drop it to a much smaller number (4?) for at least the qpg and efr32 and nrfconnect platforms.  Not sure about others.

Reply:
OK, this has now been updated in the latest commit. All except linux, darwin and android are using 4 cache spots.

Reply:
Are we missing something here?  Is this feature relevant for anything _but_ constrained platforms?

Surely real Linux implementations will achieve the same (and with much less effort) by setting Avahi cache-entries-max to something large.

Long term, it seems like we need to make this feature usable and beneficial on constrained platforms, or just abandon it.  I fully assumed that non-RTOS nodes would do this caching completely outside the sdk.

Reply:
@cecille ?

Reply:
I honestly don't know what this was added for - I'm just updating it. To me, it doesn't make sense to carry a cache. In Pankaj's refactor, he mentions removing the requirement to store the active devices and replacing that with the device addresses via a DNS cache (https://github.com/project-chip/connectedhomeip/pull/10808). I'm not sure the details of how that's meant to work, or whether TTL is being taken into account with that. 

The cache looks like it was originally written by Kamil, so it might have something to do with SRP.

@pan-apple for more info
@kkasperczyk-no who wrote the original code.

Reply:
> The cache looks like it was originally written by Kamil
@cecille what cache do you mean? The one implemented in `DnssdCache.h`? I don't remember writing any cache mechanism for Dnnsd, but I was making mdns -> dnssd refactor some time ago, so I renamed this file from `MdnsCache` to `DnssdCache` and that's why in the file history it may look like it was added by me.

Reply:
The original author of the cache is @mleisner (see https://github.com/project-chip/connectedhomeip/pull/8747).

Reply:
Ah, sorry Kamil - I just grabbed the blame and it looks like your rename tricked it.

Reply:
Given that this PR doesn't actually add the cache, but just updates it to hold all the relevant information, would it be OK with everyone if I rebase this PR and push it, then open another issue to separately consider the utility of the DNS cache and what we want to do with it?

Reply:
https://github.com/project-chip/connectedhomeip/issues/11632